diffrax/_progress_bar.py

```diff
@@ -0,0 +1,186 @@
+import abc
+
+import equinox as eqx
+import jax
+import jax.numpy as jnp
+import numpy as np
+from jax import Array
+from jax.experimental import io_callback
+from tqdm import tqdm
```

Comment:
As a stylistic convention, we tend to reserve `from` inputs for either type hints or within-package hints. I'd prefer:
```python
import jax.experimental.io_callback as io_callback
import tqdm.tqdm as tqdm
from jaxtyping import Array  # and by convention we import Array from here
```

Also, I note that this will fail if `tqdm` isn't available on the user's machine. Can we lazy-load it when `TqdmProgressBar` is instantiated for the first time?

Reply:
I agree with your comments, however if I try to replace `from jax.experimental import io_callback` by `import jax.experimental.io_callback as io_callback` I get a `ModuleNotFoundError`:
```
../diffrax/_progress_bar.py:8: in <module>
    import jax.experimental.io_callback as io_callback
E   ModuleNotFoundError: No module named 'jax.experimental.io_callback'
```
For some reason I cannot seem to be able to import functions direclty from `jax.experimental`. Maybe I'm missing something there.